sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -94,6 +94,11 @@ class ZeCall {
 };
 std::mutex ZeCall::GlobalLock;
 
+// The global mutex to get rid of calling zeEventCreate from simultaneous
+// threads in multithreading application (prohibited by Level Zero spec).
+// Without the lock, RT finishes with CL_INVALID_VALUE from piEventCreate
+static std::mutex EventCreateMutex;
```

Comment:
Is event creation a quick (of almost constant time) operation?
If so, I suggest using a spin-lock here instead of mutex to soften the performance impact.

FYI, https://stackoverflow.com/a/5870415

Reply:
Please don't use global mutexes at all, they're not trivially destructible on Windows and may lead to sporadic segfaults.